lib/core.js

```diff
@@ -47,6 +47,7 @@ declare class Object {
     static defineProperty<T>(o: any, p: any, attributes: PropertyDescriptor<T>): any;
     static entries(object: any): Array<[string, mixed]>;
     static freeze<T>(o: T): T;
+    static fromEntries<K:string, T>(entries: Array<[K, T]>): {[K]: ?T};
```

Comment:
`fromEntries` accepts any iterable, not just arrays, and the entries can be any object having 0 and 1 keys: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries.

I think this would be closer to the spec:

```suggestion
    static fromEntries<K, V>(entries: Iterable<{0: K, 1: V}>): {[K]: V};
```

Reply:
Actually... there's a limitation in Flow that doesn't allow us to use arrays when the type is defined as an object with numeric keys. I'll check this and go back to you.

Reply:
A combination of them is enough:

```javascript
static fromEntried<K, V>(entries: Iterable<[K, V] | {'0': K, '1': V}>): {[K]: V};
```